models/Asset.php

```diff
@@ -1594,10 +1586,13 @@ public function getMetadata($name = null, $language = null, $strictMatch = false
         if (is_array($metaData)) {
             foreach ($metaData as $md) {
                 $md = (array)$md;
-                if (!$raw) {
-                    $md['data'] = $convert($md);
+
+                if ((empty($language) && !$strictMatch) || ($language == $md['language'])) {
```

Comment:
```suggestion
                if ((empty($md['language']) && !$strictMatch) || ($language == $md['language'])) {
```
but still it doesn't work properly, the default case when no `$language` is set, the result should be showing all of them as formerly.